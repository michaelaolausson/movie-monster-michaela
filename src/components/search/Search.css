@media screen and (max-width: 768px) {
    form button.btn {
        width: 100%;
        margin-top: 1rem;
        padding: 1rem;
    }
    .form-group {
        flex-wrap:wrap;
    }
}