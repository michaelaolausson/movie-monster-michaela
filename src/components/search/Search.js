import React from 'react';
import { Container, Row, Col, Form, Button } from 'react-bootstrap'
import Searchresult from './Searchresult.js'
import TMDB from '../../api/TMDB'
import './Search.css';

const tmdb = new TMDB();

class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {movies: {}};
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(event) {
    this.setState({value: event.target.value});
  }
  handleSubmit(event) {
    event.preventDefault();
    tmdb.search(this.state.value, this.setMovies.bind(this))
  }
  // callback function
  setMovies( movies ) {
    this.setState({movies: movies})
  }
  render() {
    return (
      <Container>
        <Row>
          <Col sm={12} md={12} lg={6}>
            <Form onSubmit={this.handleSubmit}>
              <Form.Group className="d-flex" controlId="searchMovie">
                  <Form.Control size="lg" type="text" placeholder="Name a great movie" value={this.state.value} onChange={this.handleChange} />
                  <Button className="btn btn-primary px-5" type="submit" value="Submit"><i className="fa fa-search"></i></Button>
              </Form.Group>
            </Form>
          </Col>
          <Col sm={12}>
            <Searchresult movies={this.state.movies} />
          </Col>
        </Row>
      </Container>
    );
  }
}

export default Search;
