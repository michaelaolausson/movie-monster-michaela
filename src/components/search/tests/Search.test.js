import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow  } from 'enzyme';
import Search from '../Search.js'

describe('<Search />', () => {
    const wrapper = shallow(<Search />);
    it("should match the snapshot", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
    it('renders one search button', () => {
        const searchBtn = wrapper.find('#search-button')
        expect(searchBtn.length).toBe(1)
    })
    it('renders one search field', () => {
        const searchBtn = wrapper.find('#search-button')
        expect(searchBtn.length).toBe(1)
    })
})