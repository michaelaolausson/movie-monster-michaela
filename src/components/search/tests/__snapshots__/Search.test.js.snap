// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match the snapshot 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-lg-6 col-md-12\\"><form class=\\"\\"><div class=\\"d-flex form-group\\"><input type=\\"text\\" placeholder=\\"Name a great movie\\" value=\\"\\" id=\\"search-input\\" class=\\"form-control form-control-lg\\"/><button id=\\"search-button\\" value=\\"Submit\\" type=\\"submit\\" class=\\"btn btn-info px-5 btn btn-primary\\"><i class=\\"fa fa-search\\"></i></button></div></form></div></div><div class=\\"row\\"></div></div>"`;
